src/seismometer/_api.py

```diff
@@ -1514,3 +1516,152 @@ def __init__(self):
+    target: str,
+    score_column: str,
+    *,
+    per_context: bool = False,
+    table_only: bool = False,
+) -> HTML:
+    """
+    Generates a plot with model metrics.
+
+    Parameters
+    ----------
+    metric_generator: BinaryClassifierMetricGenerator
+        class that creates metrics for a model
+    metrics: str | list[str]
+        subset of metrics to display
+    cohort_dict : dict[str, tuple[Any]]
+        dictionary of cohort columns and values used to subselect a population for evaluation
+    target : str
+        name of the target
+    score_column : str
+        score column
+    per_context : bool, optional
+        if scores should be grouped, by default False
+
+    Returns
+    -------
+    HTML
+        an html visualization of the model evaluation metrics
+    """
+    sg = Seismogram()
+    cohort_filter = FilterRule.from_cohort_dictionary(cohort_dict)
+    data = cohort_filter.filter(sg.dataframe)
+    target_event = pdh.event_value(target)
+    target_data = FilterRule.isin(target_event, (0, 1)).filter(data)
+    return binary_classifier_metric_evaluation(
+        metric_generator,
+        metrics,
+        target_data,
+        sg.entity_keys,
+        target_event,
+        score_column,
+        sg.censor_threshold,
+        per_context,
+        sg.event_aggregation_method(target),
+        sg.predict_time,
+        table_only=table_only,
+    )
+
+
+def binary_classifier_metric_evaluation(
+    metric_generator: BinaryClassifierMetricGenerator,
+    metrics: str | list[str],
+    dataframe: pd.DataFrame,
+    entity_keys: list[str],
+    target: str,
+    score_col: str,
+    censor_threshold: int = 10,
+    per_context_id: bool = False,
+    aggregation_method: str = "max",
+    ref_time: str = None,
+    table_only: bool = False,
+) -> HTML:
+    """
+    plots common model evaluation metrics
+
+    Parameters
+    ----------
+    dataframe : pd.DataFrame
+        source data
+    entity_keys : list[str]
+        columns to use for aggregation
+    target : str
+        target column
+    score_col : str
+        score column
+    censor_threshold : int, optional
+        minimum rows to allow in a plot, by default 10
+    per_context_id : bool, optional
+        report only the max score for a given entity context, by default False
+    aggregation_method : str, optional
+        method to reduce multiple scores into a single value before calculation of performance, by default "max"
+        ignored if per_context_id is False
+    ref_time : str, optional
+        reference time column used for aggregation when per_context_id is True and aggregation_method is time-based
+
+    Returns
+    -------
+    HTML
+        Plot of model evaluation metrics
+    """
+    data = (
+        pdh.event_score(
+            dataframe, entity_keys, score=score_col, ref_event=ref_time, aggregation_method=aggregation_method
+        )
+        if per_context_id
+        else dataframe
+    )
```

Comment:
I feel like a simple
`if per_context_id: set data to first thing`
`else: set data to dataframe`

would be simpler and more readable than a ternary statement here.

Reply:
This was just staying consistent with the other calls. I've moved this to a wrapper method that does the same thing you asked.